lib/Stats.js

```diff
@@ -108,16 +119,8 @@ class Stats {
 		const showWarnings = optionOrLocalFallback(options.warnings, true);
 		const warningsFilter = optionOrFallback(options.warningsFilter, null);
 		const showPublicPath = optionOrLocalFallback(options.publicPath, !forToString);
-		const excludeModules = [].concat(optionOrFallback(options.exclude, [])).map(item => {
-			if(typeof item === "string") {
-				const regExp = new RegExp(`[\\\\/]${item.replace(/[\-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g, "\\$&")}([\\\\/]|$|!|\\?)`); // eslint-disable-line no-useless-escape
-				return ident => regExp.test(ident);
-			}
-			if(item && typeof item === "object" && typeof item.test === "function")
-				return ident => item.test(ident);
-			if(typeof item === "function")
-				return item;
-		});
+		const excludeModules = [].concat(optionOrFallback(options.exclude, [])).map(testAgainstGivenOption);
```

Comment:
Could you add the new properties to the schema?

https://github.com/webpack/webpack/blob/master/schemas/webpackOptionsSchema.json#L960

Could you also add `"additionalProperties": false,` to this stats object schema, so additional/wrong properties are disallowed? (This will yell at future PRs that add new properties to the stats options)